dev-lang/rust/rust-1.74.1-r100.ebuild

```
@@ -504,7 +530,257 @@ src_configure() {
+		"-DCMAKE_BUILD_TYPE=Release"
+	)
+
+	if [[ -z "${LLVM_CMAKE_OPTS_EXTRA}" ]]; then
+		llvm_cmake_opts+=( "${LLVM_CMAKE_OPTS_EXTRA}")
+	fi
+
+	elog "Building bootstrap llvm"
+
+	mkdir -p "${WORKDIR}/bootstrap/llvm" || die
+	pushd "${WORKDIR}/bootstrap/llvm" 2>/dev/null || die
+		cmake "${llvm_cmake_opts[@]}" "${S}/src/llvm-project/llvm" || die "Failed to configure bootstrap llvm"
+		eninja -j$(makeopts_jobs) || die "Failed to build bootstrap llvm"
+	popd 2>/dev/null || die
+}
+
+# High level steps:
+# Our system mrustc package has built stdlib for our current platform.
+# - Step 1: Use system-installed mrustc, (m)rust(c) stdlib, and minicargo to
+#	bootstrap a `cargo` and `rustc` (mrustc-stage0)
+# - Step 2: Use minicargo and the built `rustc` to build a working `sysroot`
+#			(includes `std`, `panic_unwind``, `test`, etc.) (mrustc-stage0)
+# - Step 3: Build build libs again (this time using `cargo` and `rustc`) (mrustc-stage1)
+# - Step 4: Build a `rustc` using those libs (mrustc-stage1)
+#  - Done so there's an optimised rustc arollvm_cmake_optsund (mrustc is bad at codegen)
+# - Step 5: Build `libstd` with this `rustc` (mrustc-stage2)
+#  - Needed to match ABIs
+# Stages:
+# - mrustc-stage0: mrustc-built cargo and rustc
+# - mrustc-stage1: rustc and sysroot built with mrustc-stage0
+# - mrustc-stage2: rustc from stage1 with sysroot built with stage0
+# See:
+# - https://github.com/thepowersgang/mrustc/blob/master/run_rustc/Makefile
+# - https://github.com/thepowersgang/mrustc/blob/master/TestRustcBootstrap.sh
+# - Upstream Windows .cmd files are also a good reference for early bootstrap
+mrustc_bootstrap() {
+	export RUSTC_BOOTSTRAP=1 # Possibly the only intended use of this variable in ::gentoo
+	# export these variables now and unset them at the end of the function so they don't leak
+	# into the rest of the build.
+	export CFG_COMPILER_HOST_TRIPLE="$(rust_abi)"
+	export CFG_RELEASE="${MRUSTC_RUST_VERSION}"	# Let's pretend we're 1.74.0
+	export CFG_RELEASE_CHANNEL="stable"
+	export CFG_VERSION="${MRUSTC_RUST_VERSION}-stable-mrustc"
+	export CFG_PREFIX="mrustc"
+	export CFG_LIBDIR_RELATIVE="lib"
+	export RUSTC_INSTALL_BINDIR="bin"
+	export REAL_LIBRARY_PATH_VAR="LD_LIBRARY_PATH"
+
+	# These flags are used in every invocation of our bootstrap `cargo`.
+	local cargo_flags="--target ${CFG_COMPILER_HOST_TRIPLE} -j $(makeopts_jobs) --release --verbose"
+
+	if use system-llvm; then
+		export LLVM_CONFIG="$(get_llvm_prefix)/bin/llvm-config"
+	else
+		llvm_bootstrap
+		export LLVM_CONFIG="${WORKDIR}/bootstrap/llvm/bin/llvm-config"
+	fi
+
+	# define the mrustc sysroot and common minicargo arguments.
+	local mrustc_sysroot="${BROOT}/usr/lib/rust/mrustc-${MRUSTC_VERSION}/lib/rustlib/${CFG_COMPILER_HOST_TRIPLE}/lib"
+	local minicargo_common_args=(
+		"-L ${mrustc_sysroot}"
+		"-j $(makeopts_jobs)"
+		"--vendor-dir ${S}/vendor"
+		"--manifest-overrides ${BROOT}/usr/share/mrustc-${MRUSTC_VERSION}/patches/rustc-${MRUSTC_RUST_VERSION}-overrides.toml"
+	)
```

I **think** this works better?
```suggestion
	local minicargo_common_args=(
		"-L" "${mrustc_sysroot}"
		"-j" "$(makeopts_jobs)"
		"--vendor-dir" "${S}/vendor"
		"--manifest-overrides" "${BROOT}/usr/share/mrustc-${MRUSTC_VERSION}/patches/rustc-${MRUSTC_RUST_VERSION}-overrides.toml"
	)
```